client/network/src/protocol.rs

```diff
@@ -451,13 +451,16 @@ pub struct ProtocolConfig {
 	pub roles: Roles,
 	/// Maximum number of peers to ask the same blocks in parallel.
 	pub max_parallel_downloads: u32,
+	/// Use default block announcement
+	pub default_announce_block: bool,
```

Comment:
I agree with Pierre that this feels very hacky and fragile. The documentation is not good either.